lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -582,43 +635,65 @@ static int s1ap_generate_s1_setup_response(
 int s1ap_mme_handle_ue_cap_indication(
     s1ap_state_t* state, __attribute__((unused)) const sctp_assoc_id_t assoc_id,
-    const sctp_stream_id_t stream, struct s1ap_message_s* message) {
-  ue_description_t* ue_ref_p                     = NULL;
-  S1ap_UECapabilityInfoIndicationIEs_t* ue_cap_p = NULL;
-  int rc                                         = RETURNok;
-  imsi64_t imsi64                                = INVALID_IMSI64;
+    const sctp_stream_id_t stream, S1ap_S1AP_PDU_t* pdu) {
+  ue_description_t* ue_ref_p = NULL;
+  S1ap_UECapabilityInfoIndication_t* container;
+  S1ap_UECapabilityInfoIndicationIEs_t* ie = NULL;
+  int rc                                   = RETURNok;
+  mme_ue_s1ap_id_t mme_ue_s1ap_id          = 0;
+  enb_ue_s1ap_id_t enb_ue_s1ap_id          = 0;
+  imsi64_t imsi64                          = INVALID_IMSI64;
 
   OAILOG_FUNC_IN(LOG_S1AP);
-  DevAssert(message != NULL);
-  ue_cap_p = &message->msg.s1ap_UECapabilityInfoIndicationIEs;
+  DevAssert(pdu != NULL);
+  container =
+      &pdu->choice.initiatingMessage.value.choice.UECapabilityInfoIndication;
 
-  if ((ue_ref_p = s1ap_state_get_ue_mmeid(ue_cap_p->mme_ue_s1ap_id)) == NULL) {
-    OAILOG_DEBUG(
-        LOG_S1AP,
-        "No UE is attached to this mme UE s1ap id: " MME_UE_S1AP_ID_FMT "\n",
-        (uint32_t) ue_cap_p->mme_ue_s1ap_id);
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_UECapabilityInfoIndicationIEs_t, ie, container,
+      S1ap_ProtocolIE_ID_id_MME_UE_S1AP_ID, true);
+
+  if (ie) {
+    mme_ue_s1ap_id = ie->value.choice.MME_UE_S1AP_ID;
+    if ((ue_ref_p = s1ap_state_get_ue_mmeid(mme_ue_s1ap_id)) == NULL) {
+      OAILOG_DEBUG(
+          LOG_S1AP,
+          "No UE is attached to this mme UE s1ap id: " MME_UE_S1AP_ID_FMT "\n",
+          mme_ue_s1ap_id);
+      OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+    }
+  } else {
     OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
   }
 
   s1ap_imsi_map_t* s1ap_imsi_map = get_s1ap_imsi_map();
   hashtable_uint64_ts_get(
-      s1ap_imsi_map->mme_ue_id_imsi_htbl,
-      (const hash_key_t) ue_cap_p->mme_ue_s1ap_id, &imsi64);
+      s1ap_imsi_map->mme_ue_id_imsi_htbl, (const hash_key_t) mme_ue_s1ap_id,
+      &imsi64);
+
+  S1AP_FIND_PROTOCOLIE_BY_ID(
+      S1ap_UECapabilityInfoIndicationIEs_t, ie, container,
+      S1ap_ProtocolIE_ID_id_eNB_UE_S1AP_ID, true);
+
+  if (ie) {
+    enb_ue_s1ap_id = (enb_ue_s1ap_id_t)(
+        ie->value.choice.ENB_UE_S1AP_ID & ENB_UE_S1AP_ID_MASK);
+  } else {
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
 
-  if (ue_ref_p->enb_ue_s1ap_id != ue_cap_p->eNB_UE_S1AP_ID) {
+  if (ue_ref_p->enb_ue_s1ap_id != enb_ue_s1ap_id) {
     OAILOG_DEBUG_UE(
         LOG_S1AP, imsi64,
         "Mismatch in eNB UE S1AP ID, known: " ENB_UE_S1AP_ID_FMT
         ", received: " ENB_UE_S1AP_ID_FMT "\n",
-        ue_ref_p->enb_ue_s1ap_id, (uint32_t) ue_cap_p->eNB_UE_S1AP_ID);
+        ue_ref_p->enb_ue_s1ap_id, (uint32_t)(uint32_t) enb_ue_s1ap_id);
```

Comment:
No need for double cast
```suggestion
        ue_ref_p->enb_ue_s1ap_id, (uint32_t) enb_ue_s1ap_id);
```

Reply:
Resolved